### ISSUE TITLE
Which places are .item() to be used in?

### ISSUE THREAD

**Issue by amitness** [PyTorchLightning/pytorch-lightning#2103]
Hi,

I read some of the example code using Lightning and in some places, there is usage of .item() in the *_epoch_end() methods to convert the loss to scalar. Is it needed or not? In PyTorch, I know this is used to prevent memory leaks, but I am unsure if this is done automatically in Lightning behind the scenes?

```python
    def validation_epoch_end(self, outputs):
        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
        return {'avg_val_loss': avg_loss.item()}
```

#### What's your environment?

 - OS: Linux
 - Packaging: conda

**Comment by williamFalcon** [comment]
we avoid .item() and instead use .detach()
but we do it automatically for the user

**Comment by amitness** [comment]
@williamFalcon Just to confirm, I don't need to worry about memory leaks, right? Even if I pass a tracked tensor in the log dictionary, the automatic detachment will handle it.

**Comment by williamFalcon** [comment]
yes...
https://towardsdatascience.com/7-tips-for-squeezing-maximum-performance-from-pytorch-ca4a40951259

**Comment by amitness** [comment]
@williamFalcon Thank you. Makes sense now.
